beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -178,8 +181,9 @@ impl<E: EthSpec> Network<E> {
         let config = ctx.config.clone();
         trace!("Libp2p Service starting");
         // initialise the node's ID
-        let local_keypair = utils::load_private_key(&config);
-
+        pub fn initialize_network(config: NetworkConfig, cli_args: &ArgMatches) {
+            let local_keypair = utils::load_private_key(&config, cli_args);
+        }
```

Comment:
This seems out of place, you don't have access to cli_args here. You should pass CLI data through the network config